### PR TITLE
feat(w3c/groups): allow specifying group type

### PR DESCRIPTION
**Pull request by sidvishnoi**
See https://github.com/marcoscaceres/respec.org/issues/114#issuecomment-694108975

```http
GET 200 /w3c/groups/webapps     # no clash, return 
GET 409 /w3c/groups/wot         # multiple groups with same shortname
GET 200 /w3c/groups/wot/wg      # okay
GET 200 /w3c/groups/webapps/wg  # for consistency
```

### CODE REVIEW

File: routes/w3c/groups.json [hunk 1]
```diff
@@ -69,6 +69,6 @@
     "webperfs": 58490,
     "webtiming": 76245,
     "wicg": 80485,
-    "wot-cg": 63991
+    "wot": 63991
```
Comment (sidvishnoi) [inline routes/w3c/groups.json:72]: Safe to rename. Not being used yet (as per GitHub search).

File: routes/w3c/index.js [hunk 2]
```diff
@@ -3,8 +3,8 @@ const cors = require("cors");
 
 const w3c = express.Router();
 
-w3c.options("/groups/:groupName?", cors({ methods: ["GET"] }));
-w3c.get("/groups/:groupName?", cors(), require("./group").route);
+w3c.options("/groups/:shortname?/:type?", cors({ methods: ["GET"] }));
+w3c.get("/groups/:shortname?/:type?", cors(), require("./group").route);
```
Comment (sidvishnoi) [inline routes/w3c/index.js:7]: All above is just for this :smile:

File: routes/w3c/group.js [hunk 3]
```diff
@@ -119,18 +135,37 @@ async function getPatentPolicy(activeCharterApiUrl) {
...
+      const msg = `No group with shortname: "${shortname}"${
+        requestedType ? ` and type: "${requestedType}"` : ""
+      }`;
+      throw new HTTPError(404, msg);
+    }
+    default: {
+      const suggestions = data.map(g => `"${shortname}/${g.type}"`).join(", ");
+      const hint = `Please specify one of following: ${suggestions}`;
+      const msg = `Multiple groups with shortname: "${shortname}".`;
+      throw new HTTPError(409, `${msg} ${hint}`);
```
Comment (sidvishnoi) [inline routes/w3c/group.js:168]: "409 Conflict" might not be semantically correct here :thinking: 
```suggestion
      throw new HTTPError(404, `${msg} ${hint}`);
```

File: routes/w3c/groups.json [hunk 4]
```diff
@@ -55,7 +55,7 @@
     "privacycg": 120428,
     "sensorweb": 52409,
     "svgcg": 112986,
-    "timed-text-cg": 50119,
+    "timed-text": 50119,
```
Comment (sidvishnoi) [inline routes/w3c/groups.json:58]: Would need to inform https://github.com/w3c/imsc about this one

**Review by marcoscaceres — APPROVED** [review]

File: routes/w3c/group.js [hunk 5]
```diff
@@ -24,17 +24,22 @@ const cache = new MemCache(ms("2 weeks"));
...
+  if (!shortname) {
     const data = await getAllGroupInfo();
     if (req.headers.accept.includes("text/html")) {
       return res.render("w3c/groups.js", { groups: data });
     }
     return res.json(data);
   }
 
+  if (type && !groups.hasOwnProperty(type)) {
+    return res.status(404).send(`Invalid group type: "${type}"`);
```
Comment (sidvishnoi) [inline routes/w3c/group.js:37]:
```suggestion
    return res.status(400).send(`Invalid group type: "${type}"`);
```

**Pull request merged**